cli/QuickstepClient.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ **/
+
+#include <chrono>
+#include <cstddef>
+#include <cstdio>
+#include <exception>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "cli/SingleNodeServer.grpc.pb.h"
+#include "cli/SingleNodeServer.pb.h"
+
+#include "gflags/gflags.h"
+#include "glog/logging.h"
+#include "grpc++/grpc++.h"
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+
+static bool ValidatePort(const char *flagname, std::int32_t value) {
+  if (value >= 0 && value < 32768)
+    return true;
+  printf("Invalid value for --%s: %d\n", flagname, (int) value);
+  return false;
+}
+
+DEFINE_int32(port, 3000, "Listens for TCP connection on this port.");
+DEFINE_validator(port, &ValidatePort);
+
+namespace quickstep {
+
+class QuickstepClient {
+public:
+  QuickstepClient(std::shared_ptr<Channel> channel)
+    : stub_(SingleNodeServerRequest::NewStub(channel)) {}
+
+  /**
+   * Assembles the client's payload, sends it and presents the response back from the server.
+   * @param user_query A SQL statement or command to be executed on the server.
+   * @return The text of the server's response.
+   */
+  std::string SendQuery(const std::string &user_query) {
+    QueryRequest request;
+    request.set_query(user_query);
+    QueryResponse response;
+    ClientContext context;
+
+    Status status = stub_->SendQuery(&context, request, &response);
+
+    if (status.ok()) {
+      return HandleQueryResponse(response);
+    } else {
+      LOG(WARNING) << "RPC call failed with code " << status.error_code()
+                   << " and message: " << status.error_message();
+      return "RPC failed";
+    }
+  }
+
+private:
+  /**
+   * Handle a valid response from the server.
+   * @param response A valid query response.
+   * @return The response string.
+   */
+  std::string HandleQueryResponse(QueryResponse const &response) {
```

Comment:
Use `std::string HandleQueryResponse(const QueryResponse & response) const {`.

In fact, it could be a static, non-member method..